src/dawn/Optimizer/Extents.cpp

```diff
@@ -37,6 +37,11 @@ void Extents::merge(const Extents& other) {
     extents_[i].merge(other.extents_[i]);
 }
 
+void Extents::expand(const Extents& other) {
+  for(std::size_t i = 0; i < extents_.size(); ++i)
```

Comment:
should and equal size check be here to avoid unexpected results?

Reply:
yes, indeed, good catch, today is hardcoded to 3, but in the future we want to generalize it. 
I added asserts in several places